crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -182,6 +188,188 @@ impl TransactionBlock {
         .map(TransactionBlock::try_from)
         .transpose()
     }
+
+    pub(crate) async fn multi_query(
+        db: &Db,
+        digests: Vec<Digest>,
+    ) -> Result<BTreeMap<Digest, Self>, Error> {
```

Comment:
Not sure I follow why this returns a BTreeMap with Digest as a key.

Reply:
This is related to this part of the PR description:

> **Transaction dependencies orderig**
>
>Fixing a bug with the order of transaction dependencies -- the previous implementation assumed that the order that the database returned transaction blocks would be the same as the order that transaction digests were passed in, but that is not true in general, and relying on that assumption messes up the consistency with which cursors are associated with transactions in the dependencies connection.

We cannot in general rely on the fact that the return values from the query will be in the same order as the digests passed in.  We could fix up the order within `multi_query`, but to do that would require translating it into a map by digest and then querying that map in sequence anyway.  We might as well return the map we would use for that instead, in case clients want to use it for point look-ups anyway.